PurrSoft/PurrSoft.Application/Commands/AnimalCommands/AnimalCommands.cs

```diff
@@ -0,0 +1,36 @@
+﻿using PurrSoft.Application.Common;
+using PurrSoft.Domain.Entities;
+using PurrSoft.Domain.Entities.Enums;
+
+namespace PurrSoft.Application.Commands.AnimalCommands;
+
+public class AnimalGetCommand: BaseRequest<CollectionResponse<Animal>>
+{
+
+}
+
+public class AnimalCreateCommand: BaseRequest<CommandResponse<int>>
+{
+    public AnimalType AnimalType { get; set; }
+    public string? Name { get; set; }
+    public int YearOfBirth { get; set; }
+    public string? Gender { get; set; }
+    public Boolean Sterilized { get; set; }
+    public string? ImageUrl { get; set; }
+}
+
+public class  AnimalUpdateCommand: BaseRequest<CommandResponse<Animal>>
+{
+    public Guid Id { get; set; }
+    public AnimalType AnimalType { get; set; }
+    public string? Name { get; set; }
+    public int YearOfBirth { get; set; }
+    public string? Gender { get; set; }
+    public Boolean Sterilized { get; set; }
+    public string? ImageUrl { get; set; }
+}
+
+public class AnimalDeleteCommand: BaseRequest<CommandResponse<Animal>>
```

Comment:
Again only if you see really fit to send the object back do it. If not you can just send the CommandResponse empty/with the failures you encounter. Because in controller the CommandResponse can be checked if valid , if not you can just send NotFound are any kind of http error. Or Ok if everything was alright. Most of the time you don't really need that information back on frontend. You are just interested in knowing that the action was done.